grouper/repositories/permission_grant.py

```diff
@@ -79,6 +95,13 @@ def revoke_all_service_account_grants(self, permission):
         # type: (str) -> List[ServiceAccountPermissionGrant]
         return self.repository.revoke_all_service_account_grants(permission)
 
+    def service_account_has_permission(self, service, permission):
+        # type: (str, str) -> bool
+        for grant in self.permission_grants_for_service_account(service):
```

Comment:
Guessing this is little enough of a bottleneck that it isn't worth making the optimizations of replicating the loop from `permission_grants_for_service_account` to short-circuit and have only a single loop?

Reply:
Yeah, that was my thought process. We can always make it faster later if it looks like it matters. I'm hoping before that happens we modify the graph to store the same data transfer objects, so then it's a simple loop either way.